botocore/utils.py

```diff
@@ -3255,7 +3255,9 @@ def _is_s3express_request(params):
     return endpoint_properties.get('backend') == 'S3Express'
 
 
-def _has_checksum_header(params):
+# This is not a public interface and is subject to abrupt breaking changes.
+# Any usage is not advised or supported in external code bases.
```

Comment:
Let's do this comment inside the function rather than before it.

Reply:
Updated in 009ea3d3b3c07886c07863673d3dd471c8241647. Let me know if the wording looks good.